cirq/contrib/noise_models/noise_models_test.py

```diff
@@ -157,3 +160,105 @@ def test_decay_noise_after_moment():
         cirq.measure(qubits[2], key='q2')
     ])
     _assert_equivalent_op_tree(true_noisy_program, noisy_circuit)
+
+
+# Fake calibration data for mock object.
+_CALIBRATION = {
+    'name':
+    'projects/foo/processors/fake_processor_name/calibrations/1579214873',
+    'timestamp': '2020-01-16T14:47:51Z',
+    'data': {
+        '@type':
+        'type.googleapis.com/cirq.google.api.v2.MetricsSnapshot',
+        'timestampMs':
+        '1579214873',
+        'metrics': [{
+            'name': 'xeb',
+            'targets': ['0_0', '0_1'],
+            'values': [{
+                'doubleVal': .9999
+            }]
+        }, {
+            'name': 'xeb',
+            'targets': ['0_0', '1_0'],
+            'values': [{
+                'doubleVal': .9998
+            }]
+        }, {
+            'name': 'single_qubit_rb_total_error',
+            'targets': ['q0_0'],
+            'values': [{
+                'doubleVal': .001
+            }]
+        }, {
+            'name': 'single_qubit_rb_total_error',
+            'targets': ['q0_1'],
+            'values': [{
+                'doubleVal': .002
+            }]
+        }, {
+            'name': 'single_qubit_rb_total_error',
+            'targets': ['q1_0'],
+            'values': [{
+                'doubleVal': .003
+            }]
+        }]
+    }
+}
+
+
+@mock.patch.object(discovery, 'build')
+def test_per_qubit_noise_from_data(build):
+    service = mock.Mock()
+    build.return_value = service
+    calibrations = service.projects().processors().calibrations()
+    calibrations.list().execute.return_value = ({
+        'calibrations': [_CALIBRATION]
+    })
+    eng = cirq.google.Engine(project_id='myproject')
+    calibration = eng.get_latest_calibration('fake_processor_name')
+    noise_model = simple_noise_from_calibration_metrics(calibration)
+    # Confirm that the data was polled from the mock engine.
+    assert calibrations.list.call_args[1][
+        'parent'] == 'projects/myproject/processors/fake_processor_name'
+
+    # Create the circuit and apply the noise model.
+    program = cirq.Circuit()
+    qubits = [cirq.GridQubit(0, 0), cirq.GridQubit(0, 1), cirq.GridQubit(1, 0)]
+    program.append([
+        cirq.H(qubits[0]),
+        cirq.CNOT(qubits[0], qubits[1]),
+        cirq.CNOT(qubits[0], qubits[2])
+    ])
+    program.append([
+        cirq.measure(qubits[0], key='q0'),
+        cirq.measure(qubits[1], key='q1'),
+        cirq.measure(qubits[2], key='q2')
+    ],
+                   strategy=cirq.InsertStrategy.NEW_THEN_INLINE)
+    noisy_circuit = cirq.Circuit(noise_model.noisy_moments(program, qubits))
+
+    # Insert channels explicitly to construct expected output.
+    true_noisy_program = cirq.Circuit()
```

Comment:
Rename to `expected`

Note the expression

```
cirq.Circuit(
    cirq.Moment(ops1),
    cirq.Moment(ops2),
    cirq.Moment(ops3))
```

would be a much more compact way to generate the circuit.

Reply:
Done and done.